stdNullPtr/Dynamic-Email-Generator
# Implement Controller

## Overview
Created a rest controller with a single endpoint, added centralized exception handling, common API response format, basic request validation, controller tests, and swagger page with customized documentation. Currently, there is a dummy service implementation and no real logic behind the controller's responses.

## Related Issues
Resolves #3  - Implement Controller

## Testing Instructions
1. Checkout the branch and build the project with `mvn clean package`.
2. Follow the instructions in the README, to setup certificates ("Certificates" section in README)
3. Follow the instructions in the README, to run the service locally ("Running the docker-compose bundle locally" section in README)
4. Use the following CURL commands to test the endpoint (refer to #2 for query param syntax and to #3 for examples, but note that since there is no service implementation the syntax doesn't matter yet):
   ```bash
   curl https://localhost:9443/app/v1/email/generate?str1=test&expression=doesnt%20;%20matter%20;%20yet
5. You should get your request back as a dummy response object: 
    ```json
    {
      "data": [
        "{str1=[test]}doesnt ; matter ; yet"
      ],
      "message": null
    }
